tests/unit/add-one-test.js

```diff
@@ -0,0 +1,25 @@
+var simple = require('simple-mock')
+var test = require('tape')
+
+var addOne = require('../../lib/helpers/add-one')
+
+test('add-one non-409 error', function (t) {
+  t.plan(1)
+
+  var non409Error = new Error('not a 409 error')
+  var state = {
+    db: {
+      put: simple.stub().rejectWith(non409Error)
+    }
+  }
+
+  addOne(state, {})
```

Comment:
You could use both, but I think `simple-mock` would be simpler, proxyquire is quite a bit of "magic" and it’s hard to debug when it doesn’t work.

What we do in cases like this is define an `internals = {}` object, then set the methods we want to be able to mock on that internals object. And then export the `internals` object as a property of the main function we export. You can see it in action at https://github.com/hoodiehq/hoodie-server/blob/master/lib/config/secret.js and then mocking at https://github.com/hoodiehq/hoodie-server/blob/master/test/unit/config/secret-test.js#L34. 

Does that help?